src/modules/job-manager/queue.c

```diff
@@ -547,12 +576,61 @@ static void queue_enable_cb (flux_t *h,
         flux_log_error (h, "error responding to job-manager.queue-enable");
 }
 
-static void queue_stop_allocations (struct queue *queue) {
-    if (alloc_pending_count (queue->ctx->alloc) > 0) {
+int job_in_queue (struct queue *queue,
+                  struct job *job,
+                  const char *name,
+                  bool *in_queue)
```

Comment:
This name is confusing on so many levels :smile:: `job_` prefix should be reserved for `job.[ch]`, all jobs are "in" a queue although it may be the anonymous one, or are we talking about the "alloc queue", and it sounds like a boolean test but it's not.

Further argument for preparsing `job->queue` and then this function can disappear and callers can just test for NULL and do string compares.

Reply:
as an aside to this comment, I wanted to rename the "alloc->queue", b/c the word "queue" was overused, but I couldn't come up with a better name.  Then of course we'd have to adjust the "alloc_enqueue" and "alloc_dequeue" function names too ...